pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java

```diff
@@ -128,10 +128,10 @@ private TableConfigUtils() {
           RoutingConfig.MULTI_STAGE_REPLICA_GROUP_SELECTOR_TYPE);
 
   /**
-   * @see TableConfigUtils#validate(TableConfig, Schema, String, boolean)
+   * @see TableConfigUtils#validate(TableConfig, Schema, String, boolean, boolean)
    */
   public static void validate(TableConfig tableConfig, @Nullable Schema schema) {
-    validate(tableConfig, schema, null, false);
+    validate(tableConfig, schema, null, false, false);
```

Comment:
Let's not keep adding booleans for each check type because it is hard to maintain. Suggest making them static, and set them up when starting the controller. `validate()` only takes `tableConfig`, `schema` and optional `typesToSkip`

Reply:
TableConfigUtils has static utility methods only. IMO, we should not store state as static variables in that class.

Reply:
These are always instance level/cluster level config, so it should be good to have them as `static` (basically model it as a singleton validator). We may add 2 static methods `setDisableGroovy()` and `setEnforcePoolBasedAssignment()`, and set them up in the `ControllerStarter`.